src/binary-writer.cc

```diff
@@ -389,11 +394,11 @@ static void write_expr(Context* ctx,
       break;
     case ExprType::BrTable: {
       write_opcode(&ctx->stream, Opcode::BrTable);
-      write_u32_leb128(&ctx->stream, expr->br_table.targets.size,
+      write_u32_leb128(&ctx->stream, expr->br_table.targets->size(),
                        "num targets");
       uint32_t depth;
-      for (size_t i = 0; i < expr->br_table.targets.size; ++i) {
-        depth = get_label_var_depth(ctx, &expr->br_table.targets.data[i]);
+      for (size_t i = 0; i < expr->br_table.targets->size(); ++i) {
```

Comment:
How about:
    for (...& target : expr->br_table.targets) { depth = ...(ctx, *target); ....}